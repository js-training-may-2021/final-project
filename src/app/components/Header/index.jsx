import React from 'react';
import cn from 'classnames';

import { isDarkTheme } from '@utils/themeUtils';

import Navigation from '@components/Navigation';
import Logo from '@components/Logo';

import styles from './Header.module.scss';

const Header = () => {
  const headerClasses = cn({
    [styles.header]: true,
    [styles.dark]: isDarkTheme(),
  });

  return (
    <header className={headerClasses}>
      <div className={styles.menu}>
        <Logo />
        <Navigation />
      </div>
    </header>
  );
};

export default Header;
