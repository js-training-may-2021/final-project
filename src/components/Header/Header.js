import React from 'react';
import { Link } from "react-router-dom";
import './_style.scss';

const Header = () => {
  return (
    <header className='header'>
      <nav className='navigation'>
        <ul className='nav-list'>
          <li>
            <Link to='/' className='link-style'>Main</Link>
          </li>
          <li>
            <Link to='/caughtPokemon' className='link-style'><p>Pokemon <br/>Collection</p></Link>
          </li>
        </ul>
      </nav>
      {!window.location.href.includes('caughtPokemon') && !window.location.href.includes('pokemonInfo')
        ?  <div className='main-banner'><h3 className='main-banner-text'>pokemon<br/>dex</h3></div> : null}
    </header>
  )
}

export default Header;
