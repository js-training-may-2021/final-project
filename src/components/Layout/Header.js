import { NavLink } from 'react-router-dom';

import classes from './Header.module.scss';

const Header = () => {
  return (
    <header className={classes.header}>
      <nav>
        <ul className={classes["nav-list"]}>
          <li>
            <NavLink to='/home' className={classes["nav-link"]} activeClassName={classes["active-nav-link"]}>
              All Pokemons
            </NavLink>
          </li>
          <li>
            <NavLink to='/my-pokemons' className={classes["nav-link"]} activeClassName={classes["active-nav-link"]} >
              My Pokemons
            </NavLink>
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Header;