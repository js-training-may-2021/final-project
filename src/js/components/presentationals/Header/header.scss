@import '../../../../styles/variables';
@import '../../../../styles/mixins';

.header {
  @include nav-wrap;
  background-color: map-get($colorScheme, "header");
}
