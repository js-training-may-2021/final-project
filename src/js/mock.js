export const pokemons = [
  {
    name: 'bulbasaur',
    id: 1,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'ivysaur',
    id: 2,
    isCaught: true,
    catchDate: '23 June 2021',
  },
  {
    name: 'venusaur',
    id: 3,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'charmander',
    id: 4,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'charmeleon',
    id: 5,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'charizard',
    id: 6,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'squirtle',
    id: 7,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'wartortle',
    id: 8,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'blastoise',
    id: 9,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'caterpie',
    id: 10,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'metapod',
    id: 11,
    isCaught: false,
    catchDate: null,
  }];  

export const uncaughtPokemon = pokemons[0];
export const caughtPokemon = pokemons[1];
