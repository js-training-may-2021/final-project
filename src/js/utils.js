import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';

export const POKEMONS_PER_PAGE = 10;
const START_PAGE = 1;

let currentPage = START_PAGE;

export const getCurrentPage = () => {
	return currentPage++;
};

export const AppRoute = {
  MAIN: '/',
  DETAIL: '/detail/:id',
  CAUGHT: '/caught'
};

const monthNames = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];

export const getCatchDate = () => {
  const date = new Date();

  const day = date.getDate();
  const month = monthNames[date.getMonth()];
  const year = date.getFullYear();

  return `${day} ${month} ${year}`
};

export const noop = () => {};
export const history = createMemoryHistory();
const mockStore = configureStore();
export const testStore = mockStore({});
