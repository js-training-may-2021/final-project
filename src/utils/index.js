export const firstLetterToUpperCase = (str) => {
  if (str === undefined) {
    return str;
  } else {
    return str
      .toLowerCase()
      .split(" ")
      .map((s) => s.charAt(0).toUpperCase() + s.substring(1))
      .join(" ");
  }
};

export const formatDate = (dateInput) => {
  return dateInput.toLocaleString("ru-RU");
};

export const padToThree = (number) => number <= 999 ? `00${number}`.slice(-3) : number;

export const handleErrors = (response) => {
  if (!response.ok) {
    throw Error(`${response.status}: ${response.statusText}`);
  }
  return response;
};
