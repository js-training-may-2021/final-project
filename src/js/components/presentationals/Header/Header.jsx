import React from 'react';
import Menu from '../Menu/Menu.jsx';
import Logo from '../Logo/Logo.jsx';
import './header.scss';

const Header = () => {
	return (
		<header className="page__header header">
			<nav className="header__nav-wrap">
        <Logo />
				<Menu />
			</nav>
		</header>
	);
};

export default React.memo(Header);
