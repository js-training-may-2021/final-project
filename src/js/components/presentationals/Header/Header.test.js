import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import Header from './Header.jsx';
import {AppRoute, history} from '../../../utils.js';

it(`Render Header on Main page`, () => {
  const tree = renderer
    .create(
      <Router history={history}>
        <Header />
      </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Header on Caught page`, () => {
  history.push(AppRoute.CAUGHT);

  const tree = renderer
    .create(
      <Router history={history}>
        <Header />
      </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Header on Detail page`, () => {
  history.push(AppRoute.DETAIL);
  
  const tree = renderer
    .create(
      <Router history={history}>
        <Header />
      </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
