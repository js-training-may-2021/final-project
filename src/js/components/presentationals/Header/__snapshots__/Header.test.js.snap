// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header on Caught page 1`] = `
<header
  className="page__header header"
>
  <nav
    className="header__nav-wrap"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        POKEDEX
      </a>
    </div>
    <ul
      className="header__menu menu"
    >
      <li
        className="menu__item"
      >
        <a
          aria-current={null}
          className="menu__link"
          href="/"
          onClick={[Function]}
        >
          ALL
        </a>
      </li>
      <li
        className="menu__item"
      >
        <a
          aria-current="page"
          className="menu__link  menu__link--selected"
          href="/caught"
          onClick={[Function]}
          style={Object {}}
        >
          CAUGHT
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Render Header on Detail page 1`] = `
<header
  className="page__header header"
>
  <nav
    className="header__nav-wrap"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        POKEDEX
      </a>
    </div>
    <ul
      className="header__menu menu"
    >
      <li
        className="menu__item"
      >
        <a
          aria-current={null}
          className="menu__link"
          href="/"
          onClick={[Function]}
        >
          ALL
        </a>
      </li>
      <li
        className="menu__item"
      >
        <a
          aria-current={null}
          className="menu__link"
          href="/caught"
          onClick={[Function]}
        >
          CAUGHT
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Render Header on Main page 1`] = `
<header
  className="page__header header"
>
  <nav
    className="header__nav-wrap"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        POKEDEX
      </a>
    </div>
    <ul
      className="header__menu menu"
    >
      <li
        className="menu__item"
      >
        <a
          aria-current="page"
          className="menu__link menu__link--selected"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          ALL
        </a>
      </li>
      <li
        className="menu__item"
      >
        <a
          aria-current={null}
          className="menu__link"
          href="/caught"
          onClick={[Function]}
        >
          CAUGHT
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
